File: client/src/api/types/response/Sticker.ts
import {EmojiData} from "emoji-mart";
import {Upload} from "./Upload";
import {StickerUploadMetadata} from "./StickerUploadMetadata";

export interface Sticker {
    id: string,
    stickerPackId: string,
    keywords: string[],
    emojis: EmojiData[],
    upload: Upload<StickerUploadMetadata>
}

File: sticker/src/main/kotlin/chatox/sticker/model/Sticker.kt
package chatox.sticker.model

import chatox.sticker.api.request.UpdateStickerRequest
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime

@Document
data class Sticker(
        @Id
        val id: String,

        @Indexed
        val stickerPackId: String,
        val upload: Upload<StickerUploadMetadata>,
        val keywords: List<String>,
        val emojis: List<EmojiData>,
        val createdAt: ZonedDateTime,
        val updatedAt: ZonedDateTime? = null
) {
        fun equalsTo(updateStickerRequest: UpdateStickerRequest): Boolean {
                return updateStickerRequest.keywords == this.keywords
                        && updateStickerRequest.emojis == this.emojis.map { emoji -> emoji.id }
        }
}
